src/service.rs

```diff
@@ -233,7 +233,15 @@ pub struct ServiceConfig {
 }
 
 impl ServiceConfig {
-    pub unsafe fn from_raw(raw: winsvc::QUERY_SERVICE_CONFIGW) -> Result<ServiceConfig> {
+    /// Tries to parse a `QUERY_SERVICE_CONFIGW` into Rust [`ServiceConfig`].
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the `dwServiceType` field does not represent a valid
+    /// [`ServiceStartType`].
+    pub unsafe fn from_raw(
+        raw: winsvc::QUERY_SERVICE_CONFIGW,
+    ) -> Result<ServiceConfig, TryFromIntError> {
```

Comment:
Same here, `QUERY_SERVICE_CONFIGW` is actually a struct

Reply:
Whoops. Fixed.